scripts/inner/kubestellar

```diff
@@ -154,58 +160,24 @@ while (( $# > 0 )); do
     esac
     shift
 done
-if [ "$PROVIDER_NAME" == "" ]; then
-    PROVIDER_NAME="default"
-    export PROVIDER_NAME
-fi
-if [ "$PROVIDER_NAMESPACE" == "" ]; then
-    PROVIDER_NAMESPACE=spaceprovider-${PROVIDER_NAME}
-    export PROVIDER_NAMESPACE
-fi
-if [ "$KUBECONFIG_DIR" == "" ]; then
-    KUBECONFIG_DIR="${PWD}/temp-space-config"
-    mkdir -p $KUBECONFIG_DIR
-    export KUBECONFIG_DIR
-fi
-if [ "$SM_CONFIG" == "" ]; then
-    SM_CONFIG="${KUBECONFIG_DIR}/config"
-    export SM_CONFIG
-fi
-if [ "$SM_CONTEXT" == "" ]; then
-    export SM_CONTEXT=sm-mgt
-fi
-if [ "$IN_CLUSTER" == "" ]; then
-    IN_CLUSTER=true
-    export IN_CLUSTER
-fi
-echo "KUBECONFIG_DIR=${KUBECONFIG_DIR}"
-echo "SM_CONFIG=${SM_CONFIG}"
-echo "SM_CONTEXT=${SM_CONTEXT}"
-echo "PROVIDER_NAME=${PROVIDER_NAME}"
-echo "PROVIDER_NAMESPACE=${PROVIDER_NAMESPACE}"
-echo "IN_CLUSTER=${IN_CLUSTER}"
 
 if [ "$subcommand" == "" ]; then
     echo "$0: missing a subcommand" >&2
     exit 1
 fi
 
 function ensure_espw() {
-    in_cluster=
-    if $IN_CLUSTER; then
-        in_cluster="--in-cluster"
-    fi 
-    space_config="${KUBECONFIG_DIR}/${PROVIDER_NAMESPACE}-${espw_name}"
-    kubectl-kubestellar-create-space --space-name $espw_name --provider-name $PROVIDER_NAME --sm-core-config $SM_CONFIG 
-    kubectl-kubestellar-get-config-for-space --space-name $espw_name --sm-core-config $SM_CONFIG ${in_cluster} --sm-context $SM_CONTEXT --output $space_config
-    run_dex
+    local kcs_kubeconfig="$(mktemp kcs.kubeconfig#XXXX)"
+    kubectl-kubestellar-space-ensure $espw_name $in_cluster --output-kubeconfig $kcs_kubeconfig
+    ensure_dex
     sleep 2
-    run_kb_backend
-    KUBECONFIG=$space_config kubectl apply -f "$bindir/../config/crds/"
+    ensure_kb_backend "$kcs_kubeconfig"
+    KUBECONFIG=$kcs_kubeconfig kubectl apply -f "$bindir/../config/crds/"
+    rm "$kcs_kubeconfig"
```

Comment:
Perhaps use a bash trap?

Reply:
This, too, was a deliberate choice for the sake of debuggability.